src/cljs/rems/table.cljs

```diff
@@ -385,42 +396,92 @@
                           :row row})
             (:td cell)))))
 
+(rf/reg-sub
+ ::limited-rows
+ (fn [[_ table] _]
+   [(rf/subscribe [::sorted-and-filtered-rows table])
+    (rf/subscribe [:rems.config/config])])
+ (fn [[rows config] [_ table]]
+   (if-let [max-rows (or (get-in config [:tables (:id table) :max-rows]) ; only ever up to this maximum
+                         (:max-rows table))]
+     (take max-rows rows)
+     rows)))
+
+(rf/reg-sub
+ ::pages
+ (fn [[_ table] _]
+   [(rf/subscribe [::limited-rows table])
+    (rf/subscribe [::paging table])])
+ (fn [[rows paging] [_ _table]]
+   (cond (empty? rows) 0
+         (not paging) 1
+         :else (js/Math.ceil (/ (count rows)
+                                (:page-size paging))))))
+
+(rf/reg-sub
+ ::paged-rows
+ (fn [[_ table] _]
+   [(rf/subscribe [::limited-rows table])
+    (rf/subscribe [::paging table])])
+ (fn [[rows paging] [_ table]]
+   (if (:paging? paging)
+     (->> rows
+          (drop (* (:page-size paging) (:current-page paging)))
+          (take (:page-size paging)))
+     rows)))
+
+(defn paging
+  "Paging component for paging a `rems.table/table` instance
+  which takes the same `table` parameter as this component.
+
+  See `rems.table/Table` for the `table` parameter schema."
+  [table _paging-opts]
+  (s/validate Table table)
+  (reagent/create-class
+   {:component-did-mount
+    (fn []
+      ;; reset paging on navigation to return to first page
+      (rf/dispatch [::set-paging table nil]))
+
+    :reagent-render
+    (fn [table paging-opts]
+      (let [paging @(rf/subscribe [::paging table])
+            pages @(rf/subscribe [::pages table])
+            on-change (fn [value]
+                        (rf/dispatch [::set-paging table value]))]
+        (when (:paging? paging)
+          [paging/paging-field (merge {:id (str (name (:id table)) "-paging")
+                                       :on-change on-change
+                                       :paging paging
+                                       :pages pages}
+                                      paging-opts)])))}))
+
 (defn table
```

Comment:
It's true. I don't really like using `component` as that assumes one always uses it like `table/component` or otherwise it isn't greppable because there will be many identically named components. My thinking for here was to not rename `table/table` as there would be many places to change. So instead just introduce the `table/standard` as a reasonable default variant. If we decide one way or another, we could make sure every component is then renamed the same way. At least we do need many variants of the same component for various cases.

Reply:
makes sense. although i'm more neutral towards `xxx/component` given that `xxx` is some well-defined entity, like table, then default component would be more explicit to use outside of `table` namespace. agreed though, let's stick to one naming scheme.